File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        map = {0:1}
        total = currSum = 0

        for num in nums:
            currSum += num
            diff = currSum - k

            total += map.get(diff,0)
            map[currSum] = 1 + map.get(currSum,0)
        return total

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size(), l=0, r=0;
        while (r<n) {
            nums[l] = nums[r];
            if (r+1<n && nums[r+1] == nums[l]) {
                r++;
                nums[++l] = nums[r];
            };
            while (r<n && nums[r] == nums[l]) {
                r++;
            }
            l++;
        }
        return l;
    }
};